File: 2170-count-number-of-maximum-bitwise-or-subsets/count-number-of-maximum-bitwise-or-subsets.py
class Solution(object):
    def countMaxOrSubsets(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_or = 0
        for num in nums:
            max_or |= num
        def back(i, curr):
            if i == len(nums):
                return 1 if curr == max_or else 0
            in_ = back(i + 1, curr | nums[i])
            out = back(i + 1, curr)
            return in_ + out
        return back(0, 0)

File: 2170-count-number-of-maximum-bitwise-or-subsets/count-number-of-maximum-bitwise-or-subsets.java
class Solution {
    private int max_or = 0;
    private int[] nums;

    public int countMaxOrSubsets(int[] nums) {
        this.nums = nums;
        for (int num : nums) {
            max_or |= num;
        }
        return back(0, 0);
    }

    private int back(int i, int curr) {
        if (i == nums.length) {
            return curr == max_or ? 1 : 0;
        }

        int in = back(i + 1, curr | nums[i]);
        int out = back(i + 1, curr);

        return in + out;
    }
}